iosApp/iosApp/Pages/Map/HomeMapView.swift

```diff
@@ -105,8 +105,13 @@ struct HomeMapView: View {
                  */
                 nearbyVM.selectingLocation = true
             }
-            .onAppear {
-                locationDataManager.locationFetcher.requestWhenInUseAuthorization()
+            .task {
```

Comment:
question: Do we still need to request permissions from within the map view if it is being requested as part of the onboarding flow?

Reply:
I just got the allow location popup on the location onboarding screen before I tapped any buttons, is that related to this?

Reply:
But we do need to show the popup every time the app launches if the user only selects "Allow Once"

Reply:
If you get the system popup before you tap anything, that's probably a result of this sleep not being long enough, yeah. I can try wiring it up to the `ContentViewModel` to actually wait for the onboarding state to load instead of just hoping it loads faster than some hardcoded delay.